### PR TITLE
[AppService] Add interactive way to get token for staticwebapp

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**
Let users get github personal access token interactively using https://github.com/Azure/azure-cli/pull/17826

**Testing Guide**
`az staticwebapp create --login-with-github`
`az staticwebapp reconnect --login-with-github`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by miwebst — APPROVED** [review]

**Comment by panchagnula** [comment]
@calvinsID don't see any tests added here or updated -do we not have any existing ones or at least add one?

**Review by jiansong-msft — APPROVED** [review]

**Comment by calvinsID** [comment]
> @calvinsID don't see any tests added here or updated -do we not have any existing ones or at least add one?

I'm not too sure how to add tests since the functionality requires the user to navigate to github website to enter the code

**Comment by panchagnula** [comment]
> > @calvinsID don't see any tests added here or updated -do we not have any existing ones or at least add one?
> 
> I'm not too sure how to add tests since the functionality requires the user to navigate to github website to enter the code

Hmm I see - Elle had added some Mock test when she originally did the commands . May be just re-run this to make sure they still work?

**Comment by calvinsID** [comment]
> > > @calvinsID don't see any tests added here or updated -do we not have any existing ones or at least add one?
> > 
> > 
> > I'm not too sure how to add tests since the functionality requires the user to navigate to github website to enter the code
> 
> Hmm I see - Elle had added some Mock test when she originally did the commands . May be just re-run this to make sure they still work?

Yup good call, thanks! Updated

File: src/azure-cli/azure/cli/command_modules/appservice/static_sites.py [hunk 1]
```diff
@@ -192,9 +193,13 @@ def update_staticsite_users(cmd, name, roles, authentication_provider=None, user
...
                        app_location='.', api_location='.', output_location='.github/workflows',
-                       tags=None, no_wait=False, sku='Free'):
-    if not token:
+                       tags=None, no_wait=False, sku='Free', login_with_github=False):
+    if not token and not login_with_github:
         _raise_missing_token_suggestion()
+    elif not token:
+        from ._github_oauth import get_github_access_token
+        scopes = ["admin:repo_hook", "repo", "workflow"]
+        token = get_github_access_token(cmd, scopes)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/static_sites.py:202]: What if `token and login_with_github`? `not token and not login_with_github`?

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/static_sites.py (hunk 1 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/static_sites.py:202]: Then `token` will be used and login_with_github param will be ignored

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/appservice/static_sites.py:202]: Maybe we can give a warning or add it to parameter help.

**Comment by qwordy** [comment]
@panchagnula Is it OK to merge?

**Comment by calvinsID** [comment]
@qwordy added logging message if both `token` and `--login-with-github` are provided, thanks!

**Comment by qwordy** [comment]
Code owner review required

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**